library/x509write_csr.c

```diff
@@ -85,6 +86,100 @@ int mbedtls_x509write_csr_set_extension(mbedtls_x509write_csr *ctx,
                                       critical, val, val_len);
 }
 
+int mbedtls_x509write_csr_set_subject_alternative_name(mbedtls_x509write_csr *ctx,
+                                                       const mbedtls_x509_san_list *san_list)
+{
+    int ret = 0;
+    const mbedtls_x509_san_list *cur = san_list;
+    unsigned char *buf;
+    unsigned char *p;
+    size_t len;
+    size_t buflen = 0;
+
+    /* Determine the maximum size of the SubjectAltName list */
+    while (cur != NULL) {
```

Comment:
Minor nit, but wouldn't this be clearer as a for loop? There is a clear initialization, check and increment.

Reply:
Something like:
`for(cur = san_list; cur != NULL; cur = cur->next)` ?